### PR TITLE
update ticket

### PR DESCRIPTION
**Pull request by lidorz51**

### CODE REVIEW

File: Source Code/Ticket Management System/Ticket Management System.vcxproj [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <VCProjectVersion>15.0</VCProjectVersion>
     <ProjectGuid>{BCF3B1AC-70F1-40FD-8EF0-6730B6E2D4E7}</ProjectGuid>
     <RootNamespace>TicketManagementSystem</RootNamespace>
-    <WindowsTargetPlatformVersion>10.0.16299.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>8.1</WindowsTargetPlatformVersion>
```
Comment (soryy708) [inline Source Code/Ticket Management System/Ticket Management System.vcxproj]: Don't push that, it'll mess up other peoples settings.

File: Source Code/Ticket Management System/Ticket Management System.vcxproj [hunk 2]
```diff
@@ -41,7 +41,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v141</PlatformToolset>
+    <PlatformToolset>v140</PlatformToolset>
```
Comment (soryy708) [inline Source Code/Ticket Management System/Ticket Management System.vcxproj]: Don't push that, it'll mess up other peoples settings.

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 3]
```diff
@@ -354,6 +354,129 @@ void openTickets()
 		}
 	}
 
+void UpdateTitle()
+{
+	wchar_t title[256];
+	wprintf(L"Enter a new title:");
+	wscanf(L"%s", title);
+	struct Ticket* ticketh= ticketContainer_getById(title);
+	ticket_setTitle(ticketh, title);
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: Memory leak: `ticketh` not destroyed.
Also, its title is set but nothing is really updated anywhere. Missing call to ticketContainer?

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 4]
```diff
@@ -354,6 +354,129 @@ void openTickets()
...
+}
+void UpdateTier()
+{
+	wchar_t tier[256];
+	wprintf(L"Enter a new tier:");
+	wscanf(L"%s", tier);
+	struct Ticket* ticketh = ticketContainer_getById(tier);
+	ticket_setTier(ticketh, tier);
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: כנ"ל

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 5]
```diff
@@ -354,6 +354,129 @@ void openTickets()
...
+}
+void UpdateStatus()
+{
+	wchar_t status[256];
+	wprintf(L"Enter a new status:");
+	wscanf(L"%s", status);
+	struct Ticket* ticketh = ticketContainer_getById(status);
+	ticket_setTier(ticketh, status);
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: כנ"ל

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 6]
```diff
@@ -354,6 +354,129 @@ void openTickets()
...
+}
+void UpdateStakeholders()
+{
+	wchar_t stakeholders[256];
+	wprintf(L"Enter a new stakeholders:");
+	wscanf(L"%s", stakeholders);
+	struct Ticket* ticketh = ticketContainer_getById(stakeholders);
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: ???

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 7]
```diff
@@ -354,6 +354,129 @@ void openTickets()
...
+	wscanf(L"%s", tier);
+	struct Ticket* ticketh = ticketContainer_getById(tier);
+	ticket_setTier(ticketh, tier);
+}
+void UpdateStatus()
+{
+	wchar_t status[256];
+	wprintf(L"Enter a new status:");
+	wscanf(L"%s", status);
+	struct Ticket* ticketh = ticketContainer_getById(status);
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: ???

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 8]
```diff
@@ -354,6 +354,129 @@ void openTickets()
...
+	wscanf(L"%s", title);
+	struct Ticket* ticketh= ticketContainer_getById(title);
+	ticket_setTitle(ticketh, title);
+}
+void UpdateTier()
+{
+	wchar_t tier[256];
+	wprintf(L"Enter a new tier:");
+	wscanf(L"%s", tier);
+	struct Ticket* ticketh = ticketContainer_getById(tier);
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: ???

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 9]
```diff
@@ -354,6 +354,129 @@ void openTickets()
 		}
 	}
 
+void UpdateTitle()
+{
+	wchar_t title[256];
+	wprintf(L"Enter a new title:");
+	wscanf(L"%s", title);
+	struct Ticket* ticketh= ticketContainer_getById(title);
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: ???

**Comment by soryy708** [comment]
אשמח אם תריץ ותגיד לי אם משהו פה עובד בכלל.

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 10]
```diff
@@ -354,6 +354,129 @@ void openTickets()
...
+	wprintf(L"Please input severity.\n1. Medium\n2. High\n3. Critical\n4. Urgent\n");
+	do
+	{
+		wscanf(L"%d", &severity_num);
+		if (severity_num < 1 || severity_num > 4)
+		{
+			wprintf("Invaild severity. Try again.\n");
+		}
+	} while (severity_num < 1 || severity_num > 4);
+	struct Ticket* ticketh = ticketContainer_getById(severity[severity_num]);
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: `severity[severity_num-1]` didn't you mean?

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 11]
```diff
@@ -354,6 +354,129 @@ void openTickets()
...
+}
+void UpdateNote()
+{
+
+}
+
+void UpdateTicket()
+{
+	wchar_t id[256];
+	wprintf(L"Enter id pls:");
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: pls

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 12]
```diff
@@ -354,6 +354,129 @@ void openTickets()
...
+	wscanf(L"%s", id);
+	struct Ticket* ticptr = ticketContainer_getById(id);
+	if (ticptr == NULL)
+	{
+		wprintf(L"Id Not Fuond!");
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: Fuond.

File: Source Code/Ticket Management System/supportgiver_menu.c [hunk 13]
```diff
@@ -354,6 +354,129 @@ void openTickets()
...
+		menu_addOption(update_tickets, tagsOption);
+		menu_addOption(update_tickets, titleSortOption);
+
+		while (menu_isOpen(update_tickets) && !menu_hasError(update_tickets))
+		{
+			system("CLS");
+			wprintf(L"Update Tickets\n==============\n");
+			printTicketsArray(ticket);
+			menu_tick(update_tickets);
+		}
```
Comment (soryy708) [inline Source Code/Ticket Management System/supportgiver_menu.c]: Leaking the `update_tickets ` and `ticket`.

File: Source Code/Ticket Management System/ticket.h [hunk 14]
```diff
@@ -39,6 +39,8 @@ void ticket_setSeverity(struct Ticket*, wchar_t*);
 void ticket_setDescription(struct Ticket*, wchar_t*);
 void ticket_setTier(struct Ticket*, wchar_t*);
 void ticket_setStatus(struct Ticket*, wchar_t*);
+void ticket_setStakeholders(struct Ticket*, wchar_t*);
```
Comment (soryy708) [inline Source Code/Ticket Management System/ticket.h]: Stakeholders is a vector. How will you overwrite it with a `wchar_t*`?

File: Source Code/Ticket Management System/ticket.h [hunk 15]
```diff
@@ -39,6 +39,8 @@ void ticket_setSeverity(struct Ticket*, wchar_t*);
...
+void ticket_setTags(struct Ticket*, wchar_t*);
```
Comment (soryy708) [inline Source Code/Ticket Management System/ticket.h]: Tags is a vector. How will you overwrite it with a `wchar_t*`?

**Comment by soryy708** [comment]
מיותר; אימפלמנטציה עדיפה שעובדת: https://github.com/SCE-SWE-2018-G3/ticket-management-system/pull/75

File: Source Code/Ticket Management System/Ticket Management System.vcxproj (hunk 1 above)
Comment (lidorz51) [reply Source Code/Ticket Management System/Ticket Management System.vcxproj]: ok

File: Source Code/Ticket Management System/Ticket Management System.vcxproj (hunk 2 above)
Comment (lidorz51) [reply Source Code/Ticket Management System/Ticket Management System.vcxproj]: ok

**Pull request closed without merging**